frontend/src/components/course/components/modules-cards-container/modules-cards-container.tsx

```diff
@@ -1,33 +1,50 @@
 import { AppRoute } from 'common/enums/enums';
-import { CourseModulesGetAllItemResponseDto, FC } from 'common/types/types';
+import {
+  CourseGetResponseDto,
+  CourseModulesGetAllItemResponseDto,
+  FC,
+  TaskWithModuleResponseDto,
+} from 'common/types/types';
 import { Link } from 'components/common/common';
-import { useAppSelector } from 'hooks/hooks';
 
-import { ModuleCard } from './components/module-card/module-card';
+import { ModuleCard } from './components/components';
+import { getTaskForModule } from './helpers/helpers';
 import styles from './styles.module.scss';
 
 type Props = {
+  isMentorView: boolean;
+  studentId: number;
   modules: CourseModulesGetAllItemResponseDto[];
+  tasks: TaskWithModuleResponseDto[];
+  course: CourseGetResponseDto;
 };
 
-const ModulesCardsContainer: FC<Props> = ({ modules }) => {
-  const { course } = useAppSelector((state) => state.course);
-
+const ModulesCardsContainer: FC<Props> = ({
+  isMentorView,
+  studentId,
+  modules,
+  tasks,
+  course,
+}) => {
   return (
     <ul className={styles.container}>
       {modules.map((courseModule) => {
+        const linkTo = isMentorView
+          ? `${AppRoute.STUDENTS}/${studentId}${AppRoute.COURSES}/${
+              (course as CourseGetResponseDto).id
```

Comment:
can we have helper to replace `:studentId` and `:courseId`, etc.. in AppRoute, instead of creating whole route from scratch?

Reply:
@PavelSukhinin how can I type the properties? I'm trying in the following way, but TS says I can't assert to `as const`
![image](https://user-images.githubusercontent.com/61088675/189157780-2dfaa113-35f0-4ecb-8ab8-5e09b5c1c5d0.png)

`route` is of type `AppRoute`

Reply:
Something like

```tsx
const insertUrlParams = (
  url: string,
  params: Record<string, string>,
): string => {
  return Object.entries(params).reduce((path, [param, value]) => {
    return path.replace(`:${param}`, encodeURIComponent(value));
  }, url);
};
```

Reply:
I wanted to type the params, according to received route, that's why I was stuck, thanks :)